### PR TITLE
Fixes ALL lag with the Tesla Engine, and makes it 1:1 with tg 

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
rscadd: Removes ALL lag from the tesla engine.
rscadd: The tesla engine is 1:1 with /tg/station's tesla engine.
rscadd: The tesla engine will now be biased towards moving in the direction of objects it zaps.
rscadd: The tesla has multiple bolts again.
rscadd: The tesla can lose energy balls if not kept fed.
experimental: Beam() and Orbit() have been updated to the much more optimized and better /tg/station variants.
/:cl:
# what fucking retard decided that the tesla engine needed to do the following every single tick:
# EVERY TICK, it does SPAWN(-1), and then checks EVERY SINGLE MOB, OBJ, TURF, AND AREA in a 43x43 square around the tesla engine, AND THEN ROLLS A PROB TO DECIDE IF IT WANTS TO WASTE ALL THAT PROCESSING POWER

### CODE REVIEW

**Comment by AsV9** [comment]
> changelog

we use old changelog system

**Comment by Iamgoofball** [comment]
> > > > > > > > > > > > > > > > > USING OLD CHANGELOG SYSTEM

**Comment by AsV9** [comment]
idk how the new one works, don't blame me

File: code/modules/power/tesla/energy_ball.dm [hunk 1]
```diff
@@ -29,75 +28,89 @@ var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
...
+			orbitsize -= (orbitsize / world.icon_size) * (world.icon_size * 0.25)
+
+			EB.orbit(src, orbitsize, pick(FALSE, TRUE), rand(10, 25), pick(3, 4, 5, 6, 36))
+
+	else if(energy < energy_to_lower && orbiting_balls.len)
+		energy_to_raise = energy_to_raise / 1.5
+		energy_to_lower = (energy_to_raise / 1.5) - 20
+
+		var/Orchiectomy_target = pick(orbiting_balls)
+		qdel(Orchiectomy_target)
```
Comment (MrsTonedOne) [inline code/modules/power/tesla/energy_ball.dm:110]: Reminder that this is how you properly do meme var/proc names. not that "move the basket ball" or "what_does_the_scouter_say_about_the_balls_power_level" shit above

**Comment by MrsTonedOne** [comment]
New one works via a github webhooks receiver script (php) that commits the changelog based on the body of the pr when it gets the merged event.
https://github.com/tgstation/-tg-station/blob/master/tools/github_webhook_processor.php

File: code/modules/power/tesla/energy_ball.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/power/tesla/energy_ball.dm:110]: F I G H T I N G   W O R D S 
I
G
H
T
I
N
G

W
O
R
D
S

**Comment by ShadowDeath6** [comment]
the goof is loose

**Pull request merged**